aws/data_source_aws_prefix_list.go

```diff
@@ -29,47 +32,97 @@ func dataSourceAwsPrefixList() *schema.Resource {
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
 			"filter": dataSourceFiltersSchema(),
+			"owner_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"address_family": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"max_entries": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"tags": tagsSchemaComputed(),
 		},
 	}
 }
 
 func dataSourceAwsPrefixListRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
 
 	filters, filtersOk := d.GetOk("filter")
 
-	req := &ec2.DescribePrefixListsInput{}
+	req := &ec2.DescribeManagedPrefixListsInput{}
```

Comment:
Just a very forward comment here since it will dramatically change the implementation -- rather than change the API call associated with the existing data source, which can cause issues in environments that utilize restrictive IAM permissions or with AWS "compatible" APIs that may not support the new API call, the preference would be to create a separate data source so operators can choose the appropriate solution. 👍 

It may be easiest to close this pull request and separately submit a pull request with just a new `aws_ec2_managed_prefix_list` data source that for now just handles AWS managed prefix lists. Once #14068 is merged, we can add any remaining functionality to the new data source.

Reply:
Thanks @bflad,

> may not support the new API call,

I hadn't considered this detail, and it makes the right course clear in retrospect. Indeed, the managed prefix list API-s require a different set of IAM actions.

Just to confirm, is `aws_ec2_managed_prefix_list` (as well as `aws_ec2_managed_prefix_list_entry` in the resource PR) the preferred nomenclature?

Reply:
Yes, please, and thank you regarding the naming! We can separately discuss the merits of an `_entry` resource in the other pull request.